+ source ../common/common_lib.sh
++ source /root/mugen-riscv/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Fri Feb  3 09:08:13 2023 - INFO  - Start environmental preparation.
+ DNF_INSTALL samba
+ pkgs=samba
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs samba --node 1 --tempfile ''
+ tmpfile2=/tmp/tmp4yly5w3a
+ '[' -z '' ']'
+ tmpfile=/tmp/tmp4yly5w3a
+ flag=false
++ getenforce
++ grep Enforcing
+ '[' ']'
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Fri Feb  3 09:09:39 2023 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Fri Feb  3 09:09:40 2023 - INFO  - Start testing...
+ test_execution smb.service
+ service=smb.service
++ date '+%Y-%m-%d %T'
+ log_time='2023-02-03 09:09:40'
+ test_restart smb.service
+ service=smb.service
+ systemctl restart smb.service
Job for smb.service failed because the control process exited with error code.
See "systemctl status smb.service" and "journalctl -xeu smb.service" for details.
+ CHECK_RESULT 1 0 0 'smb.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='smb.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'smb.service restart failed'
+ LOG_ERROR 'smb.service restart failed'
+ message='smb.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'smb.service restart failed'
Fri Feb  3 09:09:41 2023 - ERROR - smb.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 34'
+ message='../common/common_lib.sh line 34'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 34'
Fri Feb  3 09:09:41 2023 - ERROR - ../common/common_lib.sh line 34
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status smb.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'smb.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='smb.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'smb.service restart failed'
+ LOG_ERROR 'smb.service restart failed'
+ message='smb.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'smb.service restart failed'
Fri Feb  3 09:09:48 2023 - ERROR - smb.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 37'
+ message='../common/common_lib.sh line 37'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 37'
Fri Feb  3 09:09:48 2023 - ERROR - ../common/common_lib.sh line 37
+ return 0
+ systemctl stop smb.service
+ CHECK_RESULT 0 0 0 'smb.service stop failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='smb.service stop failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status smb.service
+ grep 'Active: inactive'
+ CHECK_RESULT 1 0 0 'smb.service stop failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='smb.service stop failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'smb.service stop failed'
+ LOG_ERROR 'smb.service stop failed'
+ message='smb.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'smb.service stop failed'
Fri Feb  3 09:09:55 2023 - ERROR - smb.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 42'
+ message='../common/common_lib.sh line 42'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 42'
Fri Feb  3 09:09:56 2023 - ERROR - ../common/common_lib.sh line 42
+ return 0
+ systemctl start smb.service
Job for smb.service failed because the control process exited with error code.
See "systemctl status smb.service" and "journalctl -xeu smb.service" for details.
+ CHECK_RESULT 1 0 0 'smb.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='smb.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'smb.service start failed'
+ LOG_ERROR 'smb.service start failed'
+ message='smb.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'smb.service start failed'
Fri Feb  3 09:09:56 2023 - ERROR - smb.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 44'
+ message='../common/common_lib.sh line 44'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 44'
Fri Feb  3 09:09:57 2023 - ERROR - ../common/common_lib.sh line 44
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status smb.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'smb.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='smb.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'smb.service start failed'
+ LOG_ERROR 'smb.service start failed'
+ message='smb.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'smb.service start failed'
Fri Feb  3 09:10:03 2023 - ERROR - smb.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 47'
+ message='../common/common_lib.sh line 47'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 47'
Fri Feb  3 09:10:04 2023 - ERROR - ../common/common_lib.sh line 47
+ return 0
+ test_enabled smb.service
+ service=smb.service
++ systemctl is-enabled smb.service
+ state=disabled
+ '[' disabled == enabled ']'
+ '[' disabled == disabled ']'
++ systemctl enable smb.service
++ grep 'Created symlink'
++ awk '{print $3}'
+ symlink_file=/etc/systemd/system/multi-user.target.wants/smb.service
+ find /etc/systemd/system/multi-user.target.wants/smb.service
/etc/systemd/system/multi-user.target.wants/smb.service
+ CHECK_RESULT 0 0 0 'smb.service enable failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='smb.service enable failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl disable smb.service
Removed /etc/systemd/system/multi-user.target.wants/smb.service.
+ find /etc/systemd/system/multi-user.target.wants/smb.service
find: ‘/etc/systemd/system/multi-user.target.wants/smb.service’: No such file or directory
+ CHECK_RESULT 1 0 1 'smb.service disable failed'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='smb.service disable failed'
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ grep -v -i 'DEBUG\|INFO\|WARNING'
+ journalctl --since '2023-02-03 09:09:40' -u smb.service
+ grep -i 'fail\|error'
Feb 03 09:09:40 openEuler-riscv64 smbd[1376]: /usr/sbin/smbd: error while loading shared libraries: libgenrand-samba4.so: cannot open shared object file: No such file or directory
Feb 03 09:09:40 openEuler-riscv64 systemd[1]: smb.service: Failed with result 'exit-code'.
Feb 03 09:09:40 openEuler-riscv64 systemd[1]: Failed to start Samba SMB Daemon.
Feb 03 09:09:56 openEuler-riscv64 smbd[1394]: /usr/sbin/smbd: error while loading shared libraries: libgenrand-samba4.so: cannot open shared object file: No such file or directory
Feb 03 09:09:56 openEuler-riscv64 systemd[1]: smb.service: Failed with result 'exit-code'.
Feb 03 09:09:56 openEuler-riscv64 systemd[1]: Failed to start Samba SMB Daemon.
+ CHECK_RESULT 0 0 1 'There is an error message for the log of smb.service'
+ actual_result=0
+ expect_result=0
+ mode=1
+ error_log='There is an error message for the log of smb.service'
+ '[' -z 0 ']'
+ '[' 1 -eq 0 ']'
+ test 0x == 0x
+ test -n 'There is an error message for the log of smb.service'
+ LOG_ERROR 'There is an error message for the log of smb.service'
+ message='There is an error message for the log of smb.service'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'There is an error message for the log of smb.service'
Fri Feb  3 09:10:13 2023 - ERROR - There is an error message for the log of smb.service
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 28'
+ message='../common/common_lib.sh line 28'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 28'
Fri Feb  3 09:10:14 2023 - ERROR - ../common/common_lib.sh line 28
+ return 0
+ systemctl start smb.service
Job for smb.service failed because the control process exited with error code.
See "systemctl status smb.service" and "journalctl -xeu smb.service" for details.
+ sed -i 's\ExecStart=/usr/sbin/smbd\ExecStart=/usr/sbin/smbd -D\g' /usr/lib/systemd/system/smb.service
+ systemctl daemon-reload
+ systemctl reload smb.service
smb.service is not active, cannot reload.
+ CHECK_RESULT 1 0 0 'smb.service reload failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='smb.service reload failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'smb.service reload failed'
+ LOG_ERROR 'smb.service reload failed'
+ message='smb.service reload failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'smb.service reload failed'
Fri Feb  3 09:10:18 2023 - ERROR - smb.service reload failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_smb.sh line 40'
+ message='oe_test_service_smb.sh line 40'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_smb.sh line 40'
Fri Feb  3 09:10:19 2023 - ERROR - oe_test_service_smb.sh line 40
+ return 0
+ systemctl status smb.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'smb.service reload causes the service status to change'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='smb.service reload causes the service status to change'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'smb.service reload causes the service status to change'
+ LOG_ERROR 'smb.service reload causes the service status to change'
+ message='smb.service reload causes the service status to change'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'smb.service reload causes the service status to change'
Fri Feb  3 09:10:20 2023 - ERROR - smb.service reload causes the service status to change
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_smb.sh line 42'
+ message='oe_test_service_smb.sh line 42'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_smb.sh line 42'
Fri Feb  3 09:10:21 2023 - ERROR - oe_test_service_smb.sh line 42
+ return 0
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Fri Feb  3 09:10:21 2023 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 8
+ test 8 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Fri Feb  3 09:10:22 2023 - ERROR - failed to execute the case.
+ exit 8
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Fri Feb  3 09:10:23 2023 - INFO  - start environment cleanup.
+ sed -i 's\ExecStart=/usr/sbin/smbd -D\ExecStart=/usr/sbin/smbd\g' /usr/lib/systemd/system/smb.service
+ systemctl daemon-reload
+ systemctl reload smb.service
smb.service is not active, cannot reload.
+ systemctl stop smb.service
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmp4yly5w3a ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmp4yly5w3a
+ '[' 0 -ne 0 ']'
+ '[' false = true ']'
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Fri Feb  3 09:10:44 2023 - INFO  - Finish environment cleanup!
